src/dos/drive_local.cpp

```diff
@@ -81,37 +87,47 @@ bool localDrive::FileOpen(DOS_File * * file,char * name,Bit32u flags) {
 
 	//Flush the buffer of handles for the same file. (Betrayal in Antara)
 	Bit8u i,drive=DOS_DRIVES;
-	localFile *lfp;
-	for (i=0;i<DOS_DRIVES;i++) {
-		if (Drives[i]==this) {
-			drive=i;
+	localFile *lfp = nullptr;
+	for (i = 0; i < DOS_DRIVES; i++) {
+		if (Drives[i] == this) {
+			drive = i;
 			break;
 		}
 	}
-	for (i=0;i<DOS_FILES;i++) {
+	for (i = 0; i < DOS_FILES; i++) {
 		if (Files[i] && Files[i]->IsOpen() && Files[i]->GetDrive()==drive && Files[i]->IsName(name)) {
 			lfp=dynamic_cast<localFile*>(Files[i]);
 			if (lfp) lfp->Flush();
 		}
 	}
 
-	FILE * hand = fopen_wrap(newname,type);
-//	Bit32u err=errno;
-	if (!hand) { 
-		if((flags&0xf) != OPEN_READ) {
-			FILE * hmm = fopen_wrap(newname,"rb");
-			if (hmm) {
-				fclose(hmm);
-				LOG_MSG("Warning: file %s exists and failed to open in write mode.\nPlease Remove write-protection",newname);
+	FILE* fhandle = fopen(newname, type);
+	// Was the file opened with the desired type?
+	if (!fhandle) {
+		// No; but do the requested flags require write-access?
+		if ((flags & 0xf) != OPEN_READ) {
+			// Yes; then maybe the local file is simply write-protected, so try again.
+			fhandle = fopen_wrap(newname, "rb");
+			if (fhandle) {
+				// Ok! so the file is protected file; update the flags to indicate a normal open.
+				flags &= ~OPEN_READWRITE;
```

Comment:
If I understand the intention of available flags correctly, then this should probably be `~(OPEN_READWRITE|OPEN_WRITE)`? Or maybe some other combination… not sure yet.

Reply:
Yes, given the above comment, we will only be entering this branch if OPEN_READWRITE was requested, so that's the only flag we want to re-apply once we successfully open the file with `"rb"`.